### PR TITLE
Autocomplete

### PR DESCRIPTION
**Pull request by michaelschuff**
This will literally code your entire project for you

### CODE REVIEW

**Comment by WyattRoss** [comment]
Run the linter, then I'll take a look and review this.

File: src/main/java/com/ezasm/gui/editor/Autocomplete.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+        // Too few chars
+        if (pos - w < 2) {
+            return;
+        } else {
+            // only recheck keywords once we've typed 3 non-space characters
+            keywords.clear();
+            String s = textField.getText();
+            Matcher m = line_finder.matcher(s);
+            while (m.find()) {
```
Comment (WyattRoss) [inline src/main/java/com/ezasm/gui/editor/Autocomplete.java:72]: Rebuilding the whole list of keywords seems inefficient especially in larger files. Try this library instead?
https://github.com/bobbylight/AutoComplete

**Review by WyattRoss — CHANGES_REQUESTED** [review]

**Comment by michaelschuff** [comment]
Ok added the changes so it's optimized a bit better. Note we still have to recheck the entire document after every press of the delete key because the functions Autocomplete uses that get called are called after the delete and we don't know what was deleted so we have no way of telling if a label was partially deleted and needs to be removed from our keywords list. ATM it doesn't have to recalculate when you just type a key or paste something so thats good at least.

**Review by TrevorBrunette — APPROVED** [review]
LGTM after suggested changes

**Review by TrevorBrunette — APPROVED** [review]

**Pull request merged**